src/gta_sw_provider.c

```
@@ -515,41 +515,29 @@ static bool generate_access_token (struct provider_instance_auth_token_t * p_aut
 {
     EVP_MD_CTX * ctx = NULL;
     bool ret = false;
+    uint8_t count = 0;
 
     /* Compute and set basic_access_token (256 bit value) */
     ctx = EVP_MD_CTX_new();
     if (NULL == ctx) {
         goto err;
     }
-    if (!EVP_DigestInit_ex(ctx, EVP_sha256(), NULL)) {
-        goto err;
-    }
-
+    count += EVP_DigestInit_ex(ctx, EVP_sha256(), NULL);
     /* Note: do not include "p_auth_token_list_item->p_next", as this pointer can change */
-    if (!EVP_DigestUpdate(ctx, &(p_auth_token_list_item->target_personality_fingerprint), sizeof(p_auth_token_list_item->target_personality_fingerprint))) {
-        goto err;
-    }
-    if (!EVP_DigestUpdate(ctx, &(p_auth_token_list_item->type), sizeof(p_auth_token_list_item->type))) {
-        goto err;
-    }
-    if (!EVP_DigestUpdate(ctx, &(p_auth_token_list_item->usage), sizeof(p_auth_token_list_item->usage))) {
-        goto err;
-    }
-    if (!EVP_DigestUpdate(ctx, &(p_auth_token_list_item->freshness), sizeof(p_auth_token_list_item->freshness))) {
-        goto err;
-    }
-    if (!EVP_DigestUpdate(ctx, &(p_auth_token_list_item->binding_personality_fingerprint), sizeof(p_auth_token_list_item->binding_personality_fingerprint))) {
-        goto err;
-    }
-    if (!EVP_DigestUpdate(ctx, &(p_auth_token_list_item->derivation_profile), sizeof(p_auth_token_list_item->derivation_profile))) {
-        goto err;
-    }
+    count += EVP_DigestUpdate(ctx, &(p_auth_token_list_item->target_personality_fingerprint), sizeof(p_auth_token_list_item->target_personality_fingerprint));
```

Maybe would be good to still fail early, like in previous approach, though now it reads better.

---

I really like this suggestion, but while implementing it, I decided to go back to the initial approach due to better readability.